### PR TITLE
[gcs] Remove not used fields resource report

### PR DESCRIPTION
**Pull request by fishbone**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/protobuf/gcs.proto [hunk 1]
```diff
@@ -408,21 +408,19 @@ message ResourcesData {
   ResourceLoad resource_load_by_shape = 7;
   // Whether this node manager is requesting global GC.
   bool should_global_gc = 8;
-  // IP address of the node.
-  string node_manager_address = 9;
```
Comment (scv119) [inline src/ray/protobuf/gcs.proto:412]: should we keep the same index so it's backward compatible?

**Review by scv119 — APPROVED** [review]

File: src/ray/protobuf/gcs.proto (hunk 1 above)
Comment (iycheng) [reply src/ray/protobuf/gcs.proto:412]: We should. But I feel right now, we don't support usage of this, it should be fine now. Definitely later if we support rolling upgrade kind of features, we should keep the same index.

Comment (iycheng) [reply src/ray/protobuf/gcs.proto:412]: We should. But I feel right now, we don't support usage of this, it should be fine now. Definitely later if we support rolling upgrade kind of features, we should keep the same index.

**Comment by iycheng** [comment]
It turned out it's used by autocaler. We probably can remove it later when doing refactoring. But close this PR for now.

**Pull request closed without merging**